internal/cli/cmd/sysdump.go

```diff
@@ -29,6 +29,9 @@ func newCmdSysdump() *cobra.Command {
 		Short: "Collects information required to troubleshoot issues with Cilium and Hubble",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if sysdumpOptions.CiliumNamespace == "" {
+				sysdumpOptions.CiliumNamespace = namespace
```

Comment:
I think we only want to set this if the `--namespace` flag was explicitly specified on the command line. Currently, it is set to `kube-system` by default if `--namespace` is not specified: https://github.com/cilium/cilium-cli/blob/d6447fb5055448eb73776b1a0dc08b83c74506ea/internal/cli/cmd/cmd.go#L69

Otherwise we'd break the auto-detection logic here: https://github.com/cilium/cilium-cli/blob/d6447fb5055448eb73776b1a0dc08b83c74506ea/sysdump/sysdump.go#L155-L162

I think we could probably use https://pkg.go.dev/github.com/spf13/pflag#FlagSet.Changed to check whether the `--namespace` flag was explicitly set and only override `sysdumpOptions.CiliumNamespace` in case it was changed.